### PR TITLE
ENT-9665: Added /opt/cfengine/build ownership and permissions promises

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: ENT-9665
Changelog: none

merge together:
https://github.com/cfengine/buildscripts/pull/1179
https://github.com/cfengine/masterfiles/pull/2562

### CODE REVIEW

File: cfe_internal/enterprise/CFE_knowledge.cf [hunk 1]
```diff
@@ -350,6 +350,19 @@ bundle agent cfe_internal_setup_knowledge
...
+        perms => mog("770", "root", $(def.cf_apache_group) ),
+        comment => concat("Top level build directorys need to be owned by root",
+                   "so that cf-reactor can run git commands there, but also",
+                   "available to cfapache for access");
+
+      "/opt/cfengine/build" -> { "ENT-9665" }
+        perms => mog("770", "root", $(def.cf_apache_group) ),
+        depth_search => recurse("inf"),
+        comment => concat("Files for build projects need to be owned by root",
+                   "so that cf-reactor can run git commands there, but also",
```
Comment (nickanderson) [inline cfe_internal/enterprise/CFE_knowledge.cf:363]: This is recursing infinitely with no file selection , so making all files 770?

**Review by nickanderson — APPROVED** [review]

**Comment by craigcomstock** [comment]
Need to test a bit more, so made a draft for now.

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9106/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9106/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9106/

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9107/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9107/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9107/

**Pull request merged**